backend/adaptive_weights.py

```
@@ -0,0 +1,197 @@
+            "severity_mapping": self.severity_mapping,
+            "duplicate_search_radius": self.duplicate_search_radius
+        }
+
+        # Ensure directory exists
+        os.makedirs(os.path.dirname(self.weights_file), exist_ok=True)
+
+        try:
+            with open(self.weights_file, 'w') as f:
+                json.dump(data, f, indent=4)
+            logger.info(f"Saved weights to {self.weights_file}")
+        except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)